src/handlers.rs

```diff
@@ -714,18 +719,31 @@ pub(crate) fn handle_compile_subcommand(
     policy: String,
     script_type: String,
 ) -> Result<serde_json::Value, Error> {
+    use bdk_wallet::miniscript::descriptor::TapTree;
+    use std::sync::Arc;
+
     let policy = Concrete::<String>::from_str(policy.as_str())?;
     let legacy_policy: Miniscript<String, Legacy> = policy
         .compile()
         .map_err(|e| Error::Generic(e.to_string()))?;
     let segwit_policy: Miniscript<String, Segwitv0> = policy
         .compile()
         .map_err(|e| Error::Generic(e.to_string()))?;
+    let taproot_policy: Miniscript<String, Tap> = policy
+        .compile()
+        .map_err(|e| Error::Generic(e.to_string()))?;
 
     let descriptor = match script_type.as_str() {
         "sh" => Descriptor::new_sh(legacy_policy),
         "wsh" => Descriptor::new_wsh(segwit_policy),
         "sh-wsh" => Descriptor::new_sh_wsh(segwit_policy),
+        "tr" => {
+            // For tr descriptors, we use a well-known unspendable key (NUMS point).
+            // This ensures the key path is effectively disabled and only script path can be used.
+            // See https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki#constructing-and-spending-taproot-outputs
+            let tree = TapTree::Leaf(Arc::new(taproot_policy));
+            Descriptor::new_tr(NUMS_UNSPENDABLE_KEY.to_string(), Some(tree))
```

Comment:
I guess the NUMS_UNSPENDABLE_KEY represents XonlyPublicKey, so it is better to parse it into an XonlyPublicKey and not used as a string (since strings are used for key placeholders) to avoid type mismatch.

Reply:
I've added validation that `NUMS_UNSPENDABLE_KEY` is parsed as `XOnlyPublicKey`.
Do you mean that nums_key should be passed as `XOnlyPublicKey` to the `new_tr` function?